### ISSUE TITLE
Client crash when crafting [Nui Crash of Doom]

### ISSUE THREAD

**Issue by dpeart** [A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System#36]
The last two happened when I added more items to the crafting queue, but the queue already had items in it and crafting.

[nwmain-crash-1706098254.dmp](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/14039036/nwmain-crash-1706098254.dmp)
[nwmain-crash-1706098254.log](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/14039037/nwmain-crash-1706098254.log)
[nwmain-crash-1706103334.dmp](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/14039038/nwmain-crash-1706103334.dmp)
[nwmain-crash-1706103334.log](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/14039039/nwmain-crash-1706103334.log)

**Comment by Samuel-IH** [comment]
Hijacking this issue and repurposing it for the nui crash bug.

**Comment by Samuel-IH** [comment]
Nui has been known to cause crashes when quickly swapping out content from the server -> client. If you have experienced a crash during crafting it is quite likely to be this issue.

While we can't fix this bug (bug in the game itself) and can't take responsibility for it, posting your crash reports here may give us more data to investigate and hand off to the NWN team.

## TLDR;
Feel free to attach your crafting-crashes in a comment here. They might get looked at.

**Comment by Samuel-IH** [comment]
From [garbfink's issue](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/issues/3#issue-2070908416):

> **Your setup** Device platform: desktop (Win, MacOS, Linux), or mobile (Android, iOS) Game version: stable release or development release
> 
> **Describe the bug** Think this possibly could be on the client side but I don't know how to double check. I was using the forge, Placed some elm in. went to examine pine which was in my inventory and a fatal error occoured.
> 
> I have logged back on and tried to recreate but all seems ok. I have included crash log and dump if that helps.
> 
> **To Reproduce** open forge, insert elm examine pine trhough radial menu.
> [nwmain-crash-1704732249.dmp](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/13864176/nwmain-crash-1704732249.dmp)
> [nwmain-crash-1704732249.log](https://github.com/A-Dawn-of-Heroes-NWN-EE-PW/New-Crafting-System/files/13864177/nwmain-crash-1704732249.log)
> 
> **Screenshots** If applicable, add screenshots to help explain your problem.

**Comment by Samuel-IH** [comment]
Closing as this issue has since been fixed in recent versions of EE
